contracts/ModuleManager.sol

```diff
@@ -0,0 +1,152 @@
+pragma solidity 0.4.23;
+import "./Module.sol";
+import "./MasterCopy.sol";
+import "./Enum.sol";
+
+
+/// @title Module Manager - A contract that manages modules that can execute transactions via this contract
+/// @author Stefan George - <stefan@gnosis.pm>
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract ModuleManager is SelfAuthorized {
+
+    event ContractCreation(address newContract);
+
+    string public constant NAME = "Module Manager";
+    string public constant VERSION = "0.0.1";
+    address public constant SENTINEL_MODULES = address(0x1);
+
+    mapping (address => address) internal modules;
+
+    /// @dev Fallback function accepts Ether transactions.
+    function ()
+        external
+        payable
+    {
+
+    }
+
+    function setupModules(address to, bytes data)
+        internal
+    {
+        modules[SENTINEL_MODULES] = SENTINEL_MODULES;
```

Comment:
Maybe do this not at contract initialization but check this once the first module is added? This way we save 20k gas for safes without any module.